flocker/control/_model.py

```diff
@@ -301,9 +347,8 @@ def __invariant__(self):
         return (True, "")
 
     hostname = field(type=unicode, factory=unicode, mandatory=True)
-    applications = pset_field(Application)
-    manifestations = field(type=PMap, initial=pmap(), factory=pmap,
-                           mandatory=True)
+    applications = pset_field(Application, optional=True)
+    manifestations = pmap_field(unicode, Manifestation, optional=True)
```

Comment:
Oh, and `optional=True` means "None is allowed". It doesn't mean you can omit the attribute (that's pyrsistent terminology, I reused it for consistency). The attribute still needs to be set.